### ISSUE TITLE
ConnectionResetError(104, 'Connection reset by peer') when compiling data set

### ISSUE THREAD

**Issue by Yutaaa76** [MuMiN-dataset/mumin-build#8]
Hello! Firstly, I want to say thank you for constructing this awesome data set.

I'm trying to follow all steps of your tutorial notebook. When compiling the data set, I always got this error. I guess this may because some replies cannot be pulled to local anymore. Anyway, it doesn't affect the compiling process, the whole dataset can still be compiled.

```
2022-03-20 06:54:13,247 [INFO] Downloading dataset
Downloading MuMiN: 100% 200M/200M [00:05<00:00, 38.6MiB/s]
2022-03-20 06:54:22,308 [INFO] Converting dataset to less compressed format
2022-03-20 06:54:51,718 [INFO] Loading dataset
2022-03-20 06:55:00,892 [INFO] Shrinking dataset
2022-03-20 06:55:01,616 [INFO] Rehydrating tweet nodes Rehydrating: 100%
5261/5261 [00:39<00:00, 123.16it/s]
2022-03-20 06:55:41,218 [INFO] Rehydrating reply nodes Rehydrating: 80%
156600/196080 [1:19:10<16:21, 40.23it/s]
2022-03-20 07:45:05,274 [ERROR] [('Connection aborted.', ConnectionResetError(104, 'Connection reset by peer'))] Error in rehydrating tweets.
The parameters used were {'expansions': 'attachments.poll_ids,attachments.media_keys,author_id,entities.mentions.username,geo.place_id,in_reply_to_user_id,referenced_tweets.id,referenced_tweets.id.author_id', 'media.fields': 'duration_ms,height,media_key,preview_image_url,type,url,width,public_metrics', 'place.fields': 'contained_within,country,country_code,full_name,geo,id,name,place_type', 'poll.fields': 'duration_minutes,end_datetime,id,options,voting_status', 'tweet.fields': 'attachments,author_id,conversation_id,created_at,entities,geo,id,in_reply_to_user_id,lang,public_metrics,possibly_sensitive,referenced_tweets,reply_settings,source,text,withheld', 'user.fields': 'created_at,description,entities,id,location,name,pinned_tweet_id,profile_image_url,protected,public_metrics,url,username,verified,withheld', 'ids': '1419766026049306630,1419179535053082626,1418866617430196228,1419818929938214918,1419819108582019073,1419759924943458306,1419751987718115333,1419815044922187779,1419813735712141319,1419746088198852616,1419670675569192969,1419813104846843904,1418874699375910914,1420165357206179841,1420054853905260545,1419750925590319108,1418149067092283397,1418153826977189891,1418062869745356801,1418054670333923328,1418048322242334720,1418735995319619589,1418904955721822218,1418738661638672385,1418738467983527939,1418739162241507331,1418926908247617543,1418736432655568898,1418753185037041667,1418737465351970819,1418751910283202560,1418738900235915268,1418739856612732935,1418739081136250888,1418736592156508165,1418737256362385415,1418737448331448320,1419028600397963276,1418739597765451779,1418739464340377602,1418738654785294337,1418737963131867139,1418736113531957251,1418738840446119947,1418737527637348353,1418737100422320130,1418748815566376963,1418740966186438658,1418738485125685249,1418736817042608133,1418749708831535106,1418738185715208198,1418763262129328137,1418738418041950208,1418739111905701898,1418740809273421826,1418739763541073920,1418737715940577284,1418738129348014080,1418738334495645705,1418740520717852678,1418740314974703622,1418736092099104769,1418735990517256193,1418735969050767361,1418709410172579843,1418705709479436293,1418710499286560768,1418713910493032449,1418699589260038146,1418763906940608518,1418699431847813125,1418701413916491779,1418700067633041409,1418699500890243072,1418868262646001664,1418698912211288064,1418703268939763713,1418699607161327616,1418699736450744327,1418702188793278469,1418761089316184067,1418709105720582148,1418698873292345345,1418730257524346882,1418712177511436290,1418700011337093128,1347720016800731136,1347248562283950081,1347238633628164098,1347382097279918080,1347262227712192514,1347233634449821699,1347235750518132738,1348752879142789120,1347360138785644544,1347267576779460608,1347554116277559298,1347475613146370048,1347231636950310912'}.
```

However, when following the tutorial and calling `dataset.to_dgl()` function with the dataset with added embeddings, I got the following error. I guess this error is due to the missing part of the data above, since the number of all claims is 2168 and there may lack all replies of one certain claim. I'm not sure about this.  

```
DGLError                                  Traceback (most recent call last)
Input In [25], in <module>
      1 if 'dgl_graph' not in globals():
----> 2     dgl_graph = dataset.to_dgl()
      3 dgl_graph

File d:\develop\environment\python\python3-8-10\lib\site-packages\mumin\dataset.py:943, in MuminDataset.to_dgl(self)
    936 '''Convert the dataset to a DGL dataset.
    937 
    938 Returns:
    939     DGLHeteroGraph:
    940         The graph in DGL format.
    941 '''
    942 logger.info('Outputting to DGL')
--> 943 return build_dgl_dataset(nodes=self.nodes, relations=self.rels)

File d:\develop\environment\python\python3-8-10\lib\site-packages\mumin\dgl.py:193, in build_dgl_dataset(nodes, relations)
    191     rev_embs = emb_to_tensor(nodes['claim'], 'reviewer_emb')
    192     tensors = (claim_embs, rev_embs)
--> 193     dgl_graph.nodes['claim'].data['feat'] = torch.cat(tensors, dim=1)
    194 else:
    195     dgl_graph.nodes['claim'].data['feat'] = claim_embs

File d:\develop\environment\python\python3-8-10\lib\site-packages\dgl\view.py:84, in HeteroNodeDataView.__setitem__(self, key, val)
     80 else:
     81     assert isinstance(val, dict) is False, \
     82         'The HeteroNodeDataView has only one node type. ' \
     83         'please pass a tensor directly'
---> 84     self._graph._set_n_repr(self._ntid, self._nodes, {key : val})

File d:\develop\environment\python\python3-8-10\lib\site-packages\dgl\heterograph.py:4118, in DGLHeteroGraph._set_n_repr(self, ntid, u, data)
   4116 nfeats = F.shape(val)[0]
   4117 if nfeats != num_nodes:
-> 4118     raise DGLError('Expect number of features to match number of nodes (len(u)).'
   4119                    ' Got %d and %d instead.' % (nfeats, num_nodes))
   4120 if F.context(val) != self.device:
   4121     raise DGLError('Cannot assign node feature "{}" on device {} to a graph on'
   4122                    ' device {}. Call DGLGraph.to() to copy the graph to the'
   4123                    ' same device.'.format(key, F.context(val), self.device))

DGLError: Expect number of features to match number of nodes (len(u)). Got 2168 and 2167 instead.
```

Is there anything I can do to solve the second error? If the second error is really caused by this reason, is there a proper way to remove this certain claim and all data related to it in the compiled dataset?

**Comment by saattrupdan** [comment]
Hi and thank you for the bug report! Could you say which version of the package you're using? I.e. the output of `import mumin; print(mumin.__version__)`?

**Comment by Yutaaa76** [comment]
> Hi and thank you for the bug report! Could you say which version of the package you're using? I.e. the output of `import mumin; print(mumin.__version__)`?

Hi! It's 1.6.1 version.

**Comment by saattrupdan** [comment]
@Yutaaa76 Hi again. I tried to reproduce the errors you're receiving, but no luck.

The rehydration issue seems to be a problem on your end, as it's a "connection reset by peer". Maybe your internet was cut off? I tried querying Twitter with the exact same parameters and tweet IDs as you, and it worked just fine here. Does this error happen every time you compile?

As for the more serious problem regarding `to_dgl`, this seems to be that there is one of the claims in your dataset which is not connected to anything. Such claims _should_ automatically be removed during compilation. Just to rule that out, can you try calling `compile` on your dataset again? This should be almost instantaneous, since your dataset is already compiled, but it should also remove such "island nodes".

**Comment by saattrupdan** [comment]
@Yutaaa76 Okay I actually managed to reproduce your error now, and I think I've fixed it in the newest version (1.6.2). Can you update your package and see if that did the trick?

**Comment by Yutaaa76** [comment]
@saattrupdan Hi! The new version runs smoothly now, thanks!
